src/dmd/astbase.d

```diff
@@ -2745,14 +2745,27 @@ struct ASTBase
                 return sizeTy;
             }();
 
-        Type cto;
-        Type ito;
-        Type sto;
-        Type scto;
-        Type wto;
-        Type wcto;
-        Type swto;
-        Type swcto;
+        static struct Mcache
+        {
+            /* These are cached values that are lazily evaluated by constOf(), immutableOf(), etc.
+             * They should not be referenced by anybody but mtype.d.
+             * They can be null if not lazily evaluated yet.
+             * Note that there is no "shared immutable", because that is just immutable
+             * The point of this is to reduce the size of each Type instance as
+             * we bank on the idea that usually only one of variants exist.
+             * It will also speed up code because these are rarely referenced and
+             * so need not be in the cache.
+             */
+            Type cto;       // MODFlags.const_
+            Type ito;       // MODFlags.immutable_
+            Type sto;       // MODFlags.shared_
+            Type scto;      // MODFlags.shared_ | MODFlags.const_
+            Type wto;       // MODFlags.wild
+            Type wcto;      // MODFlags.wildconst
+            Type swto;      // MODFlags.shared_ | MODFlags.wild
+            Type swcto;     // MODFlags.shared_ | MODFlags.wildconst
```

Comment:
This naming scheme should be expanded in a follow up PR.

I.e. ``hasSharedAndConst`` instead of ``scto``

Without those comments, it is very hard to read let alone understand in random code snippets.

Reply:
Given the various combinations, I suspect that would make it harder to read. The one letter combinations make for easily read patterns, once you look at the definitions.

Reply:
So then...
```d
void foo(shared const Foo x);
```
...the above is "fine"? But unambiguous names **in the compiler's source** would be "hard" to read?
```d
t.getMcache.cto
t.getMcache.ito
t.getMcache.sto
t.getMcache.scto
t.getMcache.wto
t.getMcache.wcto
t.getMcache.swto
t.getMcache.swcto
```
these say absolutely nothing. Scratch that, these say you're looking in some `Mcache`, where "M" is... what? Member? Memory? Meritocracy?... You're forced to look up `getMcache()`, which returns `Mcache*` (???), so you look up `Mcache` which contains the shorthand names and comments alluding to the meaning of those names via `MODFlags`. So you look up `MODFlags`... The rabbit hole is much too deep.

```d
t.qualifierCache.const_
t.qualifierCache.immutable_
t.qualifierCache.shared_
t.qualifierCache.sharedConst
t.qualifierCache.inout_
t.qualifierCache.inoutConst
t.qualifierCache.sharedInout
t.qualifierCache.sharedInoutConst
```
That code speaks, quite loudly.